mne/event.py

```diff
@@ -885,3 +886,250 @@ def concatenate_events(events, first_samps, last_samps):
+            if var[:3] in ['mag', 'meg', 'eeg', 'eog', 'ecg']:
+                val = float(val)
+            elif var in ['ncateg', 'nevent']:
+                val = int(val)
+            setattr(self, var.lower(), val)
+        self.stimsource = (
+            'Internal' if self.stimsource == '1' else 'External')
+        # collect all events and categories as instance dicts
+        self._events = self._events_from_acq_pars()
+        self._categories = self._categories_from_acq_pars()
+        # collect events and categories that are used by this setup
+        for cat in self._categories.values():
+            if cat['event']:
+                self._events[cat['event']]['in_use'] = True
+            if cat['reqevent']:
+                self._events[cat['reqevent']]['in_use'] = True
+        self._categories_in_use = (
+            [cat for cat in self._categories.values() if cat['state']])
+        self._events_in_use = (
+            [ev for ev in self._events.values() if ev['in_use']])
+        # make a mne rejection dict based on the averager parameters
+        self.reject = {'grad': self.megmax, 'mag': self.magmax,
+                       'eeg': self.eegmax, 'eog': self.eogmax,
+                       'ecg': self.ecgmax}
+
+    def __repr__(self):
+        s = '<ElektaAverager | '
+        s += 'categories: %d ' % self.ncateg
+        cats_in_use = len(self._categories_in_use)
+        s += '(%d in use), ' % cats_in_use
+        s += 'events: %d ' % self.nevent
+        evs_in_use = len(self._events_in_use)
+        s += '(%d in use), ' % evs_in_use
+        s += 'stim source: %s>' % self.stimsource
+        return s
+
+    def __getitem__(self, items):
+        return self._categories[items]
+
+    def _events_from_acq_pars(self):
+        """ Collects DACQ defined events into a dict, keyed by number
+        starting from 1. Each event is itself a dict containing the event
+        definitions. """
+        events = {}
+        # evnum = '01', '02' etc.
+        for evnum in [str(x).zfill(2) for x in range(1, self.ncateg + 1)]:
+            evdi = {}
+            for var in self.event_vars:
+                # name of DACQ variable, e.g. 'ERFeventNewBits01'
+                acq_key = 'ERFevent' + var + evnum
+                # corresponding dict key, e.g. 'newbits'
+                dict_key = var.lower()
+                val = self.acq_dict[acq_key]
+                # convert numeric values
+                if dict_key in ['newbits', 'oldbits', 'newmask', 'oldmask']:
+                    val = int(val)
+                elif dict_key in ['delay']:
+                    val = float(val)
+                evdi[dict_key] = val
+                evdi['in_use'] = False
+            # events are keyed by number starting from 1
+            key = int(evnum)
+            evdi['index'] = key
+            events[key] = evdi
+        return events
+
+    def _categories_from_acq_pars(self, all_categories=False):
```

Comment:
It uses the magic constants for detecting dacq variable names. Actually, these are more generic than the averager (you might use them to detect your HPI frequencies, for example). So maybe they could sit outside ElektaAverager. I guess they could be module level constants or constants inside the (intimately related) `_acqpars_gen()` function, whichever is preferred.